File: app/src/test/java/com/improve10x/hackathononlinelibrarymanagementsystem/Invoice.java
package com.improve10x.hackathononlinelibrarymanagementsystem;

public class Invoice {
    public Invoice() {
    }

    public Invoice(String id, String buyerDetails, String sellerDetails, String appName, String purchaseDate, String quantity, String orderId, String paymentInfo, String book) {
        this.id = id;
        this.buyerDetails = buyerDetails;
        this.sellerDetails = sellerDetails;
        this.appName = appName;
        this.purchaseDate = purchaseDate;
        this.quantity = quantity;
        this.orderId = orderId;
        this.paymentInfo = paymentInfo;
        this.book = book;
    }

    public String id;
        public String buyerDetails;
        public String sellerDetails;
        public String appName;
        public String purchaseDate;
        public String quantity;
        public String orderId;
        public String paymentInfo;
        public String book;
}

File: app/src/main/java/com/improve10x/hackathononlinelibrarymanagementsystem/invoicemanagement/Invoice.kt
package com.improve10x.hackathononlinelibrarymanagementsystem.invoicemanagement;

import com.improve10x.hackathononlinelibrarymanagementsystem.bookmanagement.Book
import com.improve10x.hackathononlinelibrarymanagementsystem.paymentmanagement.PaymentInfo
import com.improve10x.hackathononlinelibrarymanagementsystem.usermanagement.UserInf
import java.io.Serializable;

 data class Invoice (
    var id : String? = "",
    var buyerDetails : UserInf? = null,
    var sellerDetails : UserInf? = null,
    var appName : String? = "RENUKA CHALLA",
    var purchaseDate : String? = "",
    var quantity : String? = "",
    var orderId : String? = "",
    var paymentInfo : PaymentInfo? = null,
    var book : Book? = null,
) : Serializable {
   fun createNestedObject() : Invoice {
      return this.copy(
         buyerDetails = buyerDetails?.copy(
            invoices = emptyList(),
            books = emptyList()
         ),
         sellerDetails = sellerDetails?.copy(
            invoices = emptyList(),
            books = emptyList()
         )
      )
   }
}

